pyro/poutine/handlers.py

```diff
@@ -248,7 +248,14 @@ def broadcast(fn=None):
     ...     return sample
     """
     msngr = BroadcastMessenger()
-    return msngr(fn) if fn is not None else msngr
+    if fn is not None:
```

Comment:
I guess we can omit this logic until we see a real performance cost of double wrapping.